waspc/README.md

```diff
@@ -335,13 +350,16 @@ If it happens just once every so it is probably nothing to worry about. If it ha
 ### Typical Release Process
 - Update ChangeLog.md with release notes and open an PR for feedback.
 - After approval, squash and merge PR for ChangeLog.md into `main`.
-- Make sure you are on `main` and up to date locally :D and then run `./new-release 0.x.y.z`.
+- Update your local repository state to have all remote changes.
+- Backmerge `release` into `main`.
+- Merge `main` into `release`.
+- Make sure you are on `release` and then run `./new-release 0.x.y.z`.
   - This will automatically create a new commit for updating the version in waspc.cabal, tag it, and push it all.
 - Wait for CI to finish & succeed for the new tag.
   - This will automatically create a new draft release.
 - Find new draft release here: https://github.com/wasp-lang/wasp/releases and edit it with your release notes.
 - Publish the draft release when ready.
-- Publish new [docs](/web#deployment).
+- Publish new [docs](/web#deployment) from the `release` branch as well.
 - Announce new release in Discord.
```

Comment:
```suggestion
- Make sure your `release` and `main` branches are up to date with the remote (run `git fetch`).
- Update `release` with the `main` by checking out `release` and running `git merge main`. Resolve any conflicts.
- Make sure you are on `release` and then run `./new-release 0.x.y.z`.
  - This will automatically create a new commit for updating the version in waspc.cabal, tag it, and push it all.
- Wait for CI to finish & succeed for the new tag.
  - This will automatically create a new draft release.
- Find new draft release here: https://github.com/wasp-lang/wasp/releases and edit it with your release notes.
- Publish the draft release when ready.
- Publish new [docs](/web#deployment) from the `release` branch as well.
- Announce new release in Discord.
- Update `main` with the `release` by checking out `main` and running `git merge --squash release`. There should be no conflicts.
```

Maybe we should not do `--squash` in that last merge, I am not sure. `--squash` will ensure history is kept linear in `main`. But it will squash the work done on `release`. I am not sure if this is better or it is better to have merges in two directions.